### PR TITLE
Frontend consistent code style

### PR DESCRIPTION
**Pull request by mertyldrr**
### Description
This PR adds 
1) ESLint(static code analysis tool) - to detect errors and enforce best practices in the code.
2) Prettier - to make sure that the code style is consistent. 
3) Husky - run ESLint before each commit to prevent breaking commits

### Testing
1) run "npm install"
2) Make sure to install the prettier extension in VS Code:
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode
3) Set it as the default formatter in your VS Code settings. This setting can be set for all languages or by a specific language.
4) Format a few files and observe that source control is clear(no changes in terms of code style)

### CODE REVIEW

File: frontend/package-lock.json
Comment (mrahtapot) [inline frontend/package-lock.json:1]: Everything else seems fine, the only thing I wonder here is about the different versions that are not related to the changes in this pull request. This wouldn't create a problem would it? @NCEghtebas

**Comment by bikemc** [comment]
I don't think husky pre-commit is working in the way it is intended. It should scan all frontend files with eslint to prevent any errors and it should scan all files with prettier to format them and add these files to commit as formatted. I tried to commit several lines without a semicolon at the end, it committed without changing it. Rather than creating a .husky folder we can put following code to package.json file:

    "husky": {
        "hooks": {
            "pre-commit": "npx prettier --write . && npx eslint --fix . && git add -A ."
        }
    },
    
    If we use pre-commit hooks, we do not need to use VSCode or prettier extension.
